src/modules/users/infra/typeorm/repositories/UserTokensRepository.ts

```diff
@@ -0,0 +1,28 @@
+import { getRepository, Repository } from 'typeorm'
+import IUserTokensRepository from '@modules/users/repositories/IUserTokensRepository'
+import UserToken from '../entities/UserToken'
+
+class UserTokensRepository implements IUserTokensRepository {
+  private ormRepository: Repository<UserToken>
+
+  constructor() {
+    this.ormRepository = getRepository(UserToken)
+  }
+
+  public async findByToken(token: string): Promise<UserToken | undefined> {
+    const user = await this.ormRepository.findOne({ where: { token } })
+    return user
+  }
```

Comment:
É possível retornar diretamente o resultado da profime de `.findOne`, pois nesse bloco de código não tem necessidade de você esperar a execução com `await`:

```suggestion
  public findByToken(token: string): Promise<UserToken | undefined> {
    return this.ormRepository.findOne({ where: { token } })
  }
```